src/app/containers/Caption/index.jsx

```diff
@@ -1,33 +1,35 @@
 import React from 'react';
-import { node, string } from 'prop-types';
+import { objectOf, any } from 'prop-types';
 import Caption from '../../components/Figure/Caption';
-import Text from '../../components/Text';
 import { ServiceContext } from '../../contexts/ServiceContext';
 import VisuallyHiddenText from '../../components/VisuallyHiddenText';
+import Blocks from '../Blocks';
+import Fragment from '../Fragment';
+import InlineLink from '../InlineLink';
 
-const CaptionWithOffscreenText = ({ children }) => (
+const componentsToRender = { fragment: Fragment, urlLink: InlineLink };
+
+const renderText = block => {
+  const textBlocks = block.model.blocks[0].model.blocks[0].model.blocks;
```

Comment:
Do we need to defend against this? feels like a place "could not find X of undefined" could come up

Reply:
Yeah -- as discussed in person, [I reckon this work should be done alongside the refactor Chris suggested above](https://github.com/BBC-News/simorgh/pull/1033#discussion_r238358370), as both aren't exclusive to this PR, but occur in places throughout our containers.